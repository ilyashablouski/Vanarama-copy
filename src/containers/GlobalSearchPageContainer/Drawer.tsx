import Text from 'core/atoms/text/Text';
import cx from 'classnames';
import Close from 'core/assets/icons/Close';
import Icon from 'core/atoms/icon';

interface IProps {
  renderContent: any;
  isFiltersRender: boolean;
  onCloseDrawer: () => void;
  isShowDrawer: boolean;
}
const Drawer = ({
  isFiltersRender,
  onCloseDrawer,
  isShowDrawer,
  renderContent,
}: IProps) => {
  return (
    <div
      className={cx('srp-f-flyout', {
        open: isShowDrawer,
      })}
    >
      <Text tag="div" className="title">
        <span>{isFiltersRender ? 'Filter' : 'Sort'}</span>
        <Icon
          icon={<Close />}
          onClick={onCloseDrawer}
          className="close-filter"
        />
      </Text>
      <div
        className={cx('content', {
          filters: isFiltersRender,
          sort: !isFiltersRender,
        })}
      >
        {renderContent()}
      </div>
    </div>
  );
};

export default Drawer;
