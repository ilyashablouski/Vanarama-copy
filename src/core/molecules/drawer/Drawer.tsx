import React from 'react';
import Text from 'core/atoms/text/Text';
import cx from 'classnames';
import Close from 'core/assets/icons/Close';
import Icon from 'core/atoms/icon';
import Spinner from 'core/molecules/drawer/Spinner';
import { Component } from '../../../types/common';

interface IProps {
  renderContent?: Component;
  renderActions?: Component;
  isLoading?: boolean;
  onCloseDrawer: () => void;
  isShowDrawer: boolean;
  title: string;
  dataUiTestId?: string;
}
const Drawer = ({
  onCloseDrawer,
  isShowDrawer,
  renderContent,
  title,
  renderActions,
  isLoading,
  dataUiTestId,
}: IProps) => {
  return (
    <>
      <div
        className={cx('srp-f-flyout', {
          open: isShowDrawer,
        })}
      >
        <Text tag="div" className="title">
          <span>{title}</span>
          <Icon
            icon={<Close />}
            onClick={onCloseDrawer}
            className="close-filter"
            data-uitestid={`${dataUiTestId}_icon_close-filter`}
          />
        </Text>
        {renderContent || null}
        {renderActions || null}
      </div>
      <div
        role="presentation"
        onClick={onCloseDrawer}
        className={cx('srp-f-mask', {
          open: isShowDrawer,
          updating: isLoading,
        })}
        data-uitestid={`${dataUiTestId}_div_shadow-overlay`}
      >
        <Spinner />
      </div>
    </>
  );
};

export default Drawer;
