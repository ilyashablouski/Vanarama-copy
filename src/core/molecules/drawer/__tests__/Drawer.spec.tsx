import React from 'react';
import { shallow } from 'enzyme';
import Drawer from '../Drawer';

describe('<Drawer />', () => {
  it('should render correctly', () => {
    // ACT
    const wrapper = shallow(
      <Drawer title="Test" isShowDrawer onCloseDrawer={jest.fn()} />,
    );

    // ASSERT
    expect(wrapper).toMatchSnapshot();
  });
});
