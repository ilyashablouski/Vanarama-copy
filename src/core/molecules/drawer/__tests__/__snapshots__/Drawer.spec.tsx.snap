// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> should render correctly 1`] = `
<Fragment>
  <div
    className="srp-f-flyout open"
  >
    <Memo(Text)
      className="title"
      tag="div"
    >
      <span>
        Test
      </span>
      <Icon
        className="close-filter"
        data-uitestid="global-search-page-container_icon_close-filter"
        icon={<Memo />}
        onClick={[MockFunction]}
      />
    </Memo(Text)>
  </div>
  <div
    className="srp-f-mask open"
    data-uitestid="global-search-page-container_div_shadow-overlay"
    onClick={[MockFunction]}
    role="presentation"
  >
    <Memo() />
  </div>
</Fragment>
`;
